packages/vitest/src/runtime/suite.ts

```diff
@@ -273,13 +274,27 @@ function formatTitle(template: string, items: any[], idx: number) {
   const count = template.split('%').length - 1
   let formatted = util.format(template, ...items.slice(0, count))
   if (isObject(items[0])) {
-    formatted = formatted.replace(/\$([$\w_.]+)/g, (_, key) => {
-      return objectAttr(items[0], key)
-    })
+    formatted = formatted.replace(/\$([$\w_.]+)/g,
+      (_, key) => formatObject(objectAttr(items[0], key)),
+    )
   }
   return formatted
 }
 
+function formatObject(value: unknown, maxDepth = 4): string {
+  const formatFunctions = {
+    test: (val: unknown) => typeof val === 'function',
+    serialize: (val: Function) => val.toString(),
+  }
+
+  return prettyFormat(value, {
```

Comment:
I think it's not that readable having your whole object in the test tile. Maybe we should use chai's `objDisplay`?

Reply:
does vitest ship with chai? If then, I'll switch to it.

Reply:
Yes, it uses chai for assertions.

Reply:
ok. tho i think showing full object won't be much of a problem since
1) it has depth limit (currently 3, could be reduced to be lower)
2) you can only print out individual property using `$foo.bar` notation.

Reply:
Ok, I checked it myself, and `objDisplay` returns a string. Please, use this function.

Reply:
Although inspect also rendered object the same way as `objDisplay` does.

Reply:
alright, changed it to use `objDisplay`. also posted updated screenshot which used `objDisplay` at pr body.
apparently `objDisplay`'s type signature returned void because it lacked `@returns` parameter in jsdoc. filed a PR in chaijs/chai#1490.

Reply:
Great 👍🏻